### PR TITLE
`StageWidget` to monitor and control each motorized stage

### PR DESCRIPTION
**Pull request by kangz12345**
This updates some features added in #248 and introduces an individual stage controller widget without signal connection.
The next PR will be a qiwis app using this widget.

See also: #245.

Screenshot 2024-04-04 at 02 24 05

### CODE REVIEW

File: iquip/apps/stage.py [hunk 1]
```diff
@@ -8,21 +8,25 @@
 
 from sipyco.pc_rpc import Client
 from PyQt5.QtCore import QObject, Qt, pyqtSignal, pyqtSlot
+from PyQt5.QtWidgets import (
+    QAbstractSpinBox, QDoubleSpinBox, QPushButton, QWidget,
+    QVBoxLayout, QHBoxLayout,
```
Comment (BECATRUE) [inline iquip/apps/stage.py]: Isn't it better to order alphabetically?
Indeed, I don't know orderling like this is the best..

File: iquip/apps/stage.py [hunk 2]
```diff
@@ -170,3 +197,134 @@ def __getattr__(self, name: str) -> Callable:
...
+    moveTo = pyqtSignal(float)
+    moveBy = pyqtSignal(float)
+    openTarget = pyqtSignal()
+    closeTarget = pyqtSignal()
+
+    def __init__(self, parent: Optional[QWidget] = None):
+        """Extended."""
+        super().__init__(parent=parent)
+        # widgets
+        self.connectionButton = QPushButton("Open", self)
```
Comment (BECATRUE) [inline iquip/apps/stage.py:234]: Please write the docstrings about its attributes.

File: iquip/apps/stage.py [hunk 3]
```diff
@@ -170,3 +197,134 @@ def __getattr__(self, name: str) -> Callable:
...
+        self.relativeBox = QDoubleSpinBox(self)
+        self.relativeBox.setButtonSymbols(QAbstractSpinBox.NoButtons)
+        self.relativeBox.setDecimals(3)
+        self.relativeBox.setSingleStep(0.001)
+        self.relativeBox.setAlignment(Qt.AlignRight)
+        self.relativePositiveButton = QPushButton("Move +", self)
+        self.relativeNegativeButton = QPushButton("Move -", self)
+        self._inner = QWidget(self)  # except connectionButton
+        # layout
+        abosluteLayout = QVBoxLayout()
```
Comment (BECATRUE) [inline iquip/apps/stage.py]: It looks a typo!

File: iquip/apps/stage.py [hunk 4]
```diff
@@ -170,3 +197,134 @@ def __getattr__(self, name: str) -> Callable:
...
+    @pyqtSlot(bool)
+    def setConnected(self, open_: bool):
+        """Sets the current connection status.
+
+        This also changes the enabled status and the connection button text.
+        
+        Args:
+            open_: True for open, False for closed.
+        """
+        self._inner.setEnabled(open_)
```
Comment (BECATRUE) [inline iquip/apps/stage.py:292]: It looks so fancy!

**Review by BECATRUE — CHANGES_REQUESTED** [review]

File: iquip/apps/stage.py (hunk 1 above)
Comment (kangz12345) [reply iquip/apps/stage.py]: Okay, I will rearrange them!

File: iquip/apps/stage.py [hunk 5]
```diff
@@ -170,3 +196,143 @@ def __getattr__(self, name: str) -> Callable:
...
+        self.positionBox = QDoubleSpinBox(self)
+        self.positionBox.setButtonSymbols(QAbstractSpinBox.NoButtons)
+        self.positionBox.setReadOnly(True)
+        self.positionBox.setDecimals(3)
+        self.positionBox.setSuffix("mm")
+        self.positionBox.setAlignment(Qt.AlignHCenter)
+        self.absoluteBox = QDoubleSpinBox(self)
+        self.absoluteBox.setButtonSymbols(QAbstractSpinBox.NoButtons)
+        self.absoluteBox.setDecimals(3)
+        self.absoluteBox.setSingleStep(0.001)
```
Comment (BECATRUE) [inline iquip/apps/stage.py:245]: Is there any reason to set its single step even if there is no button?

Comment (kangz12345) [reply iquip/apps/stage.py:245]: We can adjust the value by scrolling mouse wheel or pressing keyboard arrow keys.
I removed the buttons to save space.

Comment (BECATRUE) [reply iquip/apps/stage.py:245]: Oh, I forgot we can adjust it without clicking the button directly.
Then, please never mind😃

**Review by BECATRUE — APPROVED** [review]
LGTM

**Pull request merged**